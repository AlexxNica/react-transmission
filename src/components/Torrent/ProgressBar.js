import React from 'react';

function ProgressBar({ torrent }) {
  return (
    <progress max="100" value="80"></progress>
  );
}

export default ProgressBar;
