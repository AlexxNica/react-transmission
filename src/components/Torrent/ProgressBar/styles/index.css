.torrentProgressBar {}

.torrentProgressBar progress[value],
.torrentProgressBar progress[value]::-webkit-progress-bar {
  -webkit-appearance: none;
  appearance: none;

  width: 100%;
  background-color: #EEE;
  border-radius: 2px;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.25) inset;
}

.progressBarSeeding {
  background-color: #269E30
}
