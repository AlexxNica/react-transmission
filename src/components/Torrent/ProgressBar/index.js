import React from 'react';
import CSSModules from 'react-css-modules';

import styles from './styles/index.css';

function ProgressBar({ torrent }) {
  return (
    <div styleName='torrentProgressBar'>
      <progress max="100" value="80" />
    </div>
  );
}

export default CSSModules(ProgressBar, styles);
