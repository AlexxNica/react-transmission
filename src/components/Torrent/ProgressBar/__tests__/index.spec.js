import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from '../';

test('ProgressBar', () => {
  const component = renderer.create(
    <ProgressBar torrent={{isStopped: true, percentDone: 0.5}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
