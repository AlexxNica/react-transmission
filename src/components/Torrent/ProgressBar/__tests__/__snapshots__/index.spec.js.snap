exports[`test ProgressBar paused 1`] = `
<div
  className="progressBar paused">
  <progress
    max="100"
    value={50} />
</div>
`;

exports[`test ProgressBar seeding 1`] = `
<div
  className="progressBar seeding">
  <progress
    max="100"
    value={50} />
</div>
`;
